bitwarden_license/test/Commercial.Core.Test/SecretsManager/Queries/ServiceAccounts/CountNewServiceAccountSlotsRequiredQueryTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Bit.Commercial.Core.SecretsManager.Queries.ServiceAccounts;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.SecretsManager.Repositories;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Commercial.Core.Test.SecretsManager.Queries.ServiceAccounts;
+
+[SutProviderCustomize]
+public class CountNewServiceAccountSlotsRequiredQueryTests
+{
+    [Theory]
+    [BitAutoData(5, 2)]
+    [BitAutoData(5, 0)]
+    [BitAutoData(5, 6)]
+    public async Task CountNewServiceAccountSlotsRequiredAsync_ReturnsCorrectCount(
+        int organizationSmServiceAccounts,
+        int currentServiceAccounts,
+        int serviceAccountsToAdd,
+        Organization organization,
+        SutProvider<CountNewServiceAccountSlotsRequiredQuery> sutProvider)
+    {
+        var availableServiceAccountSlots = Math.Max(0, organizationSmServiceAccounts - currentServiceAccounts);
+        var expectedAvailableServiceAccounts = Math.Max(0, serviceAccountsToAdd - availableServiceAccountSlots);
```

Comment:
This is repeating the logic of the sut. If the sut is wrong, your logic here is probably wrong too, and the test will pass when it shouldn't. I recommend using literal values that you know are correct.